crates/rpc-types-engine/src/payload.rs

```diff
@@ -722,9 +722,37 @@ pub enum PayloadError {
     /// Invalid payload excess blob gas.
     #[error("invalid payload excess blob gas: {0}")]
     ExcessBlobGas(U256),
+    /// Pre-shanghai Payload has withdrawals.
+    #[error("pre-Shanghai payload has withdrawals")]
+    PreShanghaiBlockWithWitdrawals,
+    /// Post-shanghai Payload does not have withdrawals.
+    #[error("post-Shanghai payload does not have withdrawals")]
+    PostShanghaiBlockWithoutWitdrawals,
     /// Pre-cancun Payload has blob transactions.
     #[error("pre-Cancun payload has blob transactions")]
     PreCancunBlockWithBlobTransactions,
+    /// Pre-cancun Payload has blob gas used.
+    #[error("pre-Cancun payload has blob gas used")]
+    PreCancunBlockWithBlobGasUsed,
+    /// Pre-cancun Payload has excess blob gas.
+    #[error("pre-Cancun payload has excess blob gas")]
+    PreCancunBlockWithExcessBlobGas,
+    /// Pre-cancun Payload has cancun fields.
+    #[error("pre-Cancun payload has Cancun fields")]
+    PreCancunWithCancunFields,
+    /// Post-cancun Payload does not have blob transactions.
+    #[error("post-Cancun payload does not have blob transactions")]
+    PostCancunBlockWithoutBlobTransactions,
+    /// Post-cancun Payload does not have blob gas used.
+    #[error("post-Cancun payload does not have blob gas used")]
```

Comment:
hmm, these error msgs are not super clear imo, because this reads as if blob gas used does not exist post cancun.

any suggestions @Rjected ?

Reply:
Looks like the only reason why we need errors for each field is to make it more clear which field is missing, so we should note that a field is missing in the error.

maybe "blob gas used missing in post-cancun payload"